src/main/java/net/fabricmc/accesswidener/AccessWidenerRemapper.java

```diff
@@ -16,66 +16,104 @@
 
 package net.fabricmc.accesswidener;
 
-import java.util.Map;
-
 import org.objectweb.asm.commons.Remapper;
 
-public final class AccessWidenerRemapper {
-	private final AccessWidener input;
-	private final String to;
-	private final Remapper remapper;
+/**
+ * Decorates a {@link AccessWidenerVisitor} with a {@link Remapper}
+ * to remap names passing through the visitor if they come from a different namespace.
+ */
+public final class AccessWidenerRemapper implements AccessWidenerVisitor {
+	private final Provider remapperProvider;
+	private final String targetNamespace;
+	private final AccessWidenerVisitor delegate;
+	// is null while no header was read or when no remapping should occur
+	private Remapper remapper;
```

Comment:
Why would this ever be null, if it doesnt need remapping dont use the AccessWidenerRemapper